integ-test/src/test/java/org/opensearch/sql/ppl/DedupCommandIT.java

```diff
@@ -35,7 +38,44 @@ public void testConsecutiveDedup() throws IOException {
         executeQuery(
             String.format(
                 "source=%s | dedup male consecutive=true | fields male", TEST_INDEX_BANK));
-    verifyDataRows(result, rows(true), rows(false), rows(true), rows(false));
+
+    // Retrieve the actual data rows
+    JSONArray dataRows = result.getJSONArray("datarows");
+
+    // Extract the rows into a list of arrays
+    List<Object[]> actualRows = new ArrayList<>();
+    for (int i = 0; i < dataRows.length(); i++) {
+      JSONArray row = dataRows.getJSONArray(i);
+      actualRows.add(new Object[] {row.get(0)});
+    }
+
+    // Check if the consecutive deduplication is working as expected
+    boolean isConsecutive = true;
+    Object previousValue = null;
+    List<Object[]> expectedRows = new ArrayList<>();
+
+    for (Object[] currentRow : actualRows) {
+      Object currentValue = currentRow[0];
+      if (previousValue == null || !currentValue.equals(previousValue)) {
+        expectedRows.add(currentRow);
+      } else {
+        isConsecutive = false;
+        break;
+      }
+      previousValue = currentValue;
+    }
```

Comment:
Let's extract as a method. And I think we want to separate the logic for checking deduplication and generating expectedRows to make the logic clear and easy to understand.

Reply:
Updated. Thanks for suggestion